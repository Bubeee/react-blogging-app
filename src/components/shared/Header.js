import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './header.css';
import { connect } from 'react-redux';
import { logout } from '../../actions';

export class Header extends Component {
  logout() {

    localStorage.setItem('token', '');
    dispatch(logoutRequest());
  }

  render() {
    const isLoggedIn = localStorage.getItem('token') != "";
    return (
      <div className="header__container">
        blooooggin
        {isLoggedIn ? (
          <Link
            className="btn btn-success btn-lg header__button"
            to="/"
            onClick={this.logout.bind(this)}
          >
            Logout
          </Link>
        ) : (
          <span>
            <Link className="btn btn-success btn-lg header__button" to="/login">
              Login
            </Link>
            <Link
              className="btn btn-success btn-lg header__button"
              to="/register"
            >
              Register
            </Link>
          </span>
        )}
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  debugger;
  return {
    logout: () => {
      dispatch(logout());
    }
  };
};

export default connect(null, mapDispatchToProps)(Header);
