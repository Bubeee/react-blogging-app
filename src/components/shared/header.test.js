import React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';
import { Header } from './Header';

import localStorage from 'jest-localstorage-mock';

enzyme.configure({ adapter: new Adapter() });

describe('Header', () => {
  beforeEach(() => {
    // values stored in tests will also be available in other tests unless you run
    // localStorage.clear();
    // // or directly reset the storage
    // localStorage.__STORE__ = {};
    // // you could also reset all mocks, but this could impact your other mocks
    // jest.resetAllMocks();
    // // or individually reset a mock used
    // localStorage.setItem.mockClear();
  });
   

  it('should render correctly', () => {
    const output = enzyme.shallow(<Header />);

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly with token in local storage', () => {
    localStorage.getItem = () => 'tokenMock';
    const output = enzyme.shallow(<Header />);

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
