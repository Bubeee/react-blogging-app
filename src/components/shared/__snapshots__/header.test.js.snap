// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="header__container"
>
  blooooggin
  <Link
    className="btn btn-success btn-lg header__button"
    onClick={[Function]}
    replace={false}
    to="/"
  >
    Logout
  </Link>
</div>
`;

exports[`Header should render correctly with token in local storage 1`] = `
<div
  className="header__container"
>
  blooooggin
  <Link
    className="btn btn-success btn-lg header__button"
    onClick={[Function]}
    replace={false}
    to="/"
  >
    Logout
  </Link>
</div>
`;
